lib/resolv.rb

```diff
@@ -152,6 +152,23 @@ def each_name(address)
     }
   end
 
+  private def default_resolvers # :nodoc:
+    resolvers = [Hosts.new, DNS.new]
+
+    # macOS supports multiple DNS resolvers via additional configs
+    # (e.g. local domain resolvers, VPNs, etc.)
+    # ref: `man 5 resolver` on macOS/darwin
+    if /darwin/ =~ RUBY_PLATFORM && Dir.exist?('/etc/resolver')
+      Dir.each_child('/etc/resolver') do |filename|
+        resolver = DNS::Config.parse_resolv_conf("/etc/resolver/#{filename}")
+        resolver[:search] = [filename] unless resolver[:search]
```

Comment:
IIUC, the config filename implicitly specifies `domain` option, which is the key for DNS routing and is a different thing than `search` option or the search suffix.